platform/security/certificate/security-crl-generator/src/main/java/org/codice/ddf/security/crl/generator/CrlGenerator.java

```diff
@@ -133,36 +133,39 @@ public synchronized void run() {
       return;
     }
 
-    FileBackedOutputStream fileBackedOutputStream = null;
-    try {
-      // Read the response content and get the byte source
-      ByteSource byteSource;
-      try (InputStream entityStream = (InputStream) entity) {
-        fileBackedOutputStream = new FileBackedOutputStream(FILE_BACKED_STREAM_THRESHOLD);
-        IOUtils.copy(entityStream, fileBackedOutputStream);
-        fileBackedOutputStream.close();
-        byteSource = fileBackedOutputStream.asByteSource();
-      }
+    try (FileBackedOutputStream fileBackedOutputStream =
```

Comment:
Maybe Instead of having the 2 nested try, can we just have this try and remove the others or at least the one on line 141? e.g.
````
    FileBackedOutputStream outputStream = null;
    try (FileBackedOutputStream fileBackedOutputStream =
            new FileBackedOutputStream(FILE_BACKED_STREAM_THRESHOLD);
        InputStream entityStream = (InputStream) entity) {
      outputStream = fileBackedOutputStream;
      // Read the response content and get the byte source
      IOUtils.copy(entityStream, fileBackedOutputStream);
      fileBackedOutputStream.close();
      ByteSource byteSource = fileBackedOutputStream.asByteSource();

      File crlFile = getCrlFile(byteSource);
      // Verify that the CRL is valid
      if (!crlIsValid(byteSource)) {
        postErrorEvent("An error occurred while validating the CRL.");
        return;
      }
      writeCrlToFile(byteSource, crlFile);
    } catch (Exception e) {
      LOGGER.warn("Unable to copy CRL to local CRL. {}", e.getMessage());
      postErrorEvent("An error occurred while downloading the CRL.");
    } finally {
      // Cleanup temp file
      if (outputStream != null) {
        try {
          outputStream.reset();
        } catch (IOException e) {
          LOGGER.warn("Error occurred while deleting the temporary file. {}", e.getMessage());
        }
      }
    }
````

Reply:
I combined the 2 try-with-resources. I decided to keep the try statements that starts on line 142 and 159 as they seems to address a specific possible failures with their own unique log messages. If there is a way to consolidate them while still keeping their unique warnings that I may be missing let me know.

Reply:
Ok sounds good. One last question. Since we now have both `InputStream entityStream`  and `fileBackedOutputStream` in the try-with-resources, do we still need to have that `finally` that does the `fileBackedOutputStream.reset()`? Since can have case where fileBackedOutputStream was successfully, but entityStream caused and exception?

Reply:
I think it should stay to ensure that the file thats used by the stream is reset properly. I don't see any problem with it being reset if just the entityStream fails other than it is unnecessary for that specific case.

Reply:
Sorry had a typo, I meant to say don't we still need to have the `finally` block perform that `fileBackedOutputStream.reset()` in scenario stated? Because as of now if that scenario occurs, the `fileBackedOutputStream.reset()` will not be hit, just the `catch` on L167. Essentially moving the `finally` where it used to be, so will always be run, instead of having it inside the first Try.

Reply:
Since the `fileBackedOutputStream` is declared in the `try-with-resources` it is already closed by the time it would get to the `finally` block there so it would no longer exist. It should be fine because the reset is there to handle an error reading data into the `fileBackedOutputStream`. If it fails to create the stream at all then I believe there should be nothing that needs to be reset as it was not successfully created in the first place and the closing is done by the `try-with-resources`.